File: 01_projects/code/java-playwright/cbaps-dawms/basic-pom/src/test/java/com/cbaps/pages/CBAPSDashboardPage.java
package com.cbaps.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.cbaps.library.PlaywrightManager;

/**
 * CBAPSDashboardPage: Main dashboard for CBAPS application
 * Provides navigation to key CBAPS workflows like Requisition creation
 */
public class CBAPSDashboardPage {

	private static final Logger log = LoggerFactory.getLogger(CBAPSDashboardPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator createRequisitionBtn;

	public CBAPSDashboardPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Locator for Create Requisition button
		this.createRequisitionBtn = page.locator("button:has-text('Create Requisition')");

		// Stability anchor: wait for dashboard to load
		pwm.waitVisible("text=CBAPS Dashboard");
		log.info("CBAPS Dashboard Page loaded successfully.");
	}

	/**
	 * Navigate to Create Requisition page
	 * @return RequisitionPage
	 */
	public RequisitionPage goToCreateRequisition() {
		pwm.click(createRequisitionBtn);
		log.info("Clicked 'Create Requisition' button.");
		return new RequisitionPage(page, pwm);
	}
}

File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/cbaps/pages/cbaps_dashboard_page.py
"""
CBAPSDashboardPage: Main dashboard for CBAPS application
Provides navigation to key CBAPS workflows like Requisition creation
"""

import logging
from playwright.sync_api import Page, Locator
from cbaps.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class CBAPSDashboardPage:
    """
    CBAPSDashboardPage represents the CBAPS main dashboard.
    This is the Python equivalent of the Java CBAPSDashboardPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize CBAPSDashboardPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.create_requisition_btn: Locator = page.locator(
            "button:has-text('Create Requisition')"
        )
        
        # Stability anchor: wait for dashboard to load
        pwm.wait_visible("text=CBAPS Dashboard")
        logger.info("CBAPS Dashboard Page loaded successfully.")
    
    def go_to_create_requisition(self) -> 'RequisitionPage':
        """
        Navigate to Create Requisition page
        
        Returns:
            RequisitionPage instance
        """
        from cbaps.pages.requisition_page import RequisitionPage
        
        self.pwm.click(self.create_requisition_btn)
        logger.info("Clicked 'Create Requisition' button.")
        return RequisitionPage(self.page, self.pwm)
